pkg/cloudprovider/aws/fleet/packing/podresources.go

```diff
@@ -0,0 +1,60 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package packing
+
+import (
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+type sortablePods []*v1.Pod
+
+func (pods sortablePods) Len() int {
+	return len(pods)
+}
+
+func (pods sortablePods) Swap(i, j int) {
+	pods[i], pods[j] = pods[j], pods[i]
+}
+
+type byResourceRequested struct{ sortablePods }
+
+func (r byResourceRequested) Less(a, b int) bool {
+	cpuPodA := calculateCPURequested(r.sortablePods[a])
+	cpuPodB := calculateCPURequested(r.sortablePods[b])
+	if cpuPodA.Equal(cpuPodB) {
+		// check for memory
+		memPodA := calculateMemoryRequested(r.sortablePods[a])
+		memPodB := calculateMemoryRequested(r.sortablePods[b])
+		return memPodA.MilliValue() < memPodB.MilliValue()
+	}
+	return cpuPodA.MilliValue() < cpuPodB.MilliValue()
+}
+
+func calculateCPURequested(pod *v1.Pod) resource.Quantity {
```

Comment:
If you don't use the GetResources function here, look at this PR. https://github.com/awslabs/karpenter/commit/7e651ee19200115afe2f20ca6f3a71ba25a7ca01

IIUC this won't actually return the right CPU. Same for the similar memory function.